Fix return type for deltas function

lloydk opened the pull request:
The inferred type generated by Typescript for the result of the deltas function was incorrect. The `space` property had a type of `string | ColorSpace` when it should have been `ColorSpace`.

A new type called `DeltasReturn` has been added to fix the issue.

@LeaVerou not sure why we're returning the `space` and the `spaceId`. Can `spaceId` be removed?

LeaVerou commented:
> @LeaVerou not sure why we're returning the `space` and the `spaceId`. Can `spaceId` be removed?

Search the codebase for `spaceId` and if you find no compelling reason, I'm fine to remove it.

lloydk commented:
> > @LeaVerou not sure why we're returning the `space` and the `spaceId`. Can `spaceId` be removed?
> 
> Search the codebase for `spaceId` and if you find no compelling reason, I'm fine to remove it.

The `deltas()` tests used it so I changed the tests and removed the property from the return type.

MysteryBlokHed reviewed (approved):
LGTM
